packages/wrangler/src/deploy/deploy.ts

```diff
@@ -168,6 +168,60 @@ function errIsStartupErr(err: unknown): err is ParseError & { code: 10021 } {
 	return false;
 }
 
+export const validateRoutes = (
+	routes: Route[],
+	hasExperimentalAssets: boolean
+) => {
+	const invalidRoutes: { [route: string]: string[] } = {};
+	for (const route of routes) {
+		if (typeof route !== "string" && route.custom_domain) {
+			if (route.pattern.includes("*")) {
+				invalidRoutes[route.pattern] = [
+					...(invalidRoutes[route.pattern] ?? []),
+					`Wildcard operators (*) are not allowed in Custom Domains`,
+				];
+			}
+			if (route.pattern.includes("/")) {
+				invalidRoutes[route.pattern] = [
+					...(invalidRoutes[route.pattern] ?? []),
+					`Paths are not allowed in Custom Domains`,
+				];
+			}
+		} else {
+			if (
+				typeof route === "string" &&
+				hasExperimentalAssets &&
+				route.includes("/") &&
+				!route.endsWith("/*")
+			) {
+				invalidRoutes[route] = [
+					...(invalidRoutes[route] ?? []),
+					`Routes with paths (except for the wildcard /*) are not allowed in Workers + Assets projects`,
+				];
+			}
+			if (
+				typeof route === "object" &&
+				hasExperimentalAssets &&
+				route.pattern.includes("/") &&
+				!route.pattern.endsWith("/*")
+			) {
+				invalidRoutes[route.pattern] = [
+					...(invalidRoutes[route.pattern] ?? []),
+					`Routes with paths (except for the wildcard /*) are not allowed in Workers + Assets projects`,
+				];
+			}
```

Comment:
Is this handling routes with a path AND ending in `/*` correctly? eg `/some/path/*` – is that allowed or not?

Reply:
nope that should be allowed whoops - fixed